kcidb/unittest.py

```diff
@@ -85,7 +85,7 @@ def assertExecutes(self, stdin, name, *args,
         if result.returncode != status:
             errors.append(f"Expected exit status {status}, "
                           f"got {result.returncode}")
-        if not re.fullmatch(stdout_re, result.stdout, re.DOTALL):
+        if not re.match(stdout_re, result.stdout, re.DOTALL):
```

Comment:
This is side-stepping the problem. The use of `fullmatch()` is intentional here. Something that should be documented in the function's docstring, though. Please revert this change, and if you can figure out how, document that the regular expressions are matched fully.